examples/layers/rendering/src/sector_layout.dart

```diff
@@ -200,6 +200,14 @@ class SectorChildListParentData extends SectorParentData with ContainerParentDat
 class RenderSectorWithChildren extends RenderDecoratedSector with ContainerRenderObjectMixin<RenderSector, SectorChildListParentData> {
   RenderSectorWithChildren(BoxDecoration? decoration) : super(decoration);
 
+  @override
+  void dispose() {
+    visitChildren((RenderObject child) {
+      child.dispose();
```

Comment:
This contradicts the documentation on dispose that you should only dispose children that you created yourself?

Reply:
This particular render object does create its own children. We'd get an assertion error in the tests otherwise when the element tried to dispose them. It's just kind of confusing to me as to exactly where this happens...

Reply:
I don't think it does create its own children. They are added from the outside here: https://github.com/flutter/flutter/blob/e7b7ebc066c1b2a5aa5c19f8961307427e0142a6/examples/layers/rendering/custom_coordinate_systems.dart#L13

Reply:
is there a way to enumerate just the children that the selector created itself? Would help clarify why the children are disposed and avoid bugs if say  someone added a subclass that added more children.

Reply:
@goderbauer is right, but the test that's failing is actually failing from SectorApp rather than this file. I'm looking into it.